cider/tests/functionality/CiderHashJoinTest.cpp

```diff
@@ -238,6 +238,180 @@ class CiderOneToManySeqJoinTest : public CiderJoinTestBase {
+  }
+
+  void resetHashTable() override {
+    ArrowArray* build_array = nullptr;
+    ArrowSchema* build_schema = nullptr;
+    QueryArrowDataGenerator::generateBatchByTypes(
+        build_schema,
+        build_array,
+        100,
+        {"r_bigint", "r_int", "r_double", "r_float"},
+        {CREATE_SUBSTRAIT_TYPE(I64),
+         CREATE_SUBSTRAIT_TYPE(I32),
+         CREATE_SUBSTRAIT_TYPE(Fp64),
+         CREATE_SUBSTRAIT_TYPE(Fp32)},
+        {3, 3, 3, 3},
+        GeneratePattern::Random,
+        -30,
+        30);
+
+    build_table_.reset(new CiderBatch(
+        build_schema, build_array, std::make_shared<CiderDefaultAllocator>()));
+    duckDbQueryRunner_.createTableAndInsertArrowData(
+        build_table_name_, build_table_ddl_, {build_table_});
+  }
+};
+
+// TODO: (spevenhe) to be deprecated, now some features like left join and is null is not
+// supported yet
+// TODO: (spevenhe) add some corner case which will lead to build baseline join table
+// TODO: (spevenhe) add tests when OR and AND fully supported
+// TODO: (spevenhe) add col_int = col_bigint, col_float < 50 etc when CAST support
+// TODO: (spevenhe) join on string
+// TODO: (spevenhe) join on date
+#define INNER_HASH_JOIN_TEST_UNIT_FOR_ARROW(                                            \
+    TEST_CLASS, UNIT_NAME, PROJECT, COLUMN_A, JOIN_COMPARISON_OPERATOR)                 \
+  TEST_F(TEST_CLASS, UNIT_NAME) {                                                       \
+    assertJoinQueryRowEqualForArrowFormatAndReset(                                      \
+        "SELECT " #PROJECT " from table_probe JOIN table_hash ON l_" #COLUMN_A          \
+        " " #JOIN_COMPARISON_OPERATOR " r_" #COLUMN_A "");                              \
+    /*FILTER ON PROBE TABLE'S COLUMN WHICH IS ALSO IN JOIN CONDITION*/                  \
+    assertJoinQueryRowEqualForArrowFormatAndReset(                                      \
+        "SELECT " #PROJECT " from table_probe JOIN table_hash ON l_" #COLUMN_A          \
+        " " #JOIN_COMPARISON_OPERATOR " r_" #COLUMN_A " WHERE l_" #COLUMN_A " <  50 "); \
+    /*FILTER ON BUILD TABLE'S COLUMN WHICH IS ASLO IN JOIN CONDITION*/                  \
+    assertJoinQueryRowEqualForArrowFormatAndReset(                                      \
+        "SELECT " #PROJECT " from table_probe JOIN table_hash ON l_" #COLUMN_A          \
+        " " #JOIN_COMPARISON_OPERATOR " r_" #COLUMN_A " WHERE r_" #COLUMN_A " <  50 "); \
+    /*AVOID RECYCLE HASHTABLE*/                                                         \
+    assertJoinQueryRowEqualForArrowFormatAndReset(                                      \
+        "SELECT " #PROJECT " from table_probe INNER JOIN table_hash ON l_" #COLUMN_A    \
+        " " #JOIN_COMPARISON_OPERATOR " r_" #COLUMN_A "");                              \
+  }
+
+INNER_HASH_JOIN_TEST_UNIT_FOR_ARROW(CiderArrowOneToOneSeqNoNullJoinTest, ArrowOneToOneSeqNoNullJoinTest, *, int, =)  // NOLINT
```

Comment:
do we need `CiderArrowOneToOneSeqNullableJoinTest` ?

Reply:
I think CiderArrowOneToManyRandomNullJoinTest is enougth

Reply:
we'd better to test `hash_join_idx_nullable_cider`

Reply:
Ffor RandomNull, it will go to perfecthashtable and test hash_join_idx_nullable_cider.  Before using bitmap to represent null,  the *min-value-null* will lead to baselinehashtbale.

Reply:
added perfect hash, one-to-one type test. See CiderArrowOneToOneSeqNullableJoinTest